CONTRIBUTING.md

```diff
@@ -1,22 +1,56 @@
+1. Be clear and definitive when composing issues. For bug reports, include detailed error messages and environment description. For features, include a clear scope and acceptance criteria.
+1. Since we have different projects under the same monorepo, use labels to denote areas that the issue relates to:
+  1. `tracee-ebpf`
+  2. `tracee-rules`
+  3. `libbpfgo`
+  4. `signatures`
+  5.  If non of the labels is relevant don't add any (usually for top-level issues)
+1. We use the following labels to describe the type of issue:
+  1. `bug`
+  2. `good-first-issue`
+1. Self-assign or request assignment for issues you intend to work on. Don't work on an issue assigned to someone else without checking with them first and reassigning.
 
 ## Pull Requests
 
-- Every Pull Request should have an associated Issue unless it is a trivial fix.
-- Your PR is more likely to be accepted if it focuses on just one change.
-- Describe what the PR does. There's no convention enforced, but please try to be concise and descriptive. Treat the PR description as a commit message. Titles that start with "fix"/"add"/"improve"/"remove" are good examples.
-- There's no need to add or tag reviewers, if your PR is left unattended for too long, you can add a comment to bring it up to attention, optionally "@" mention one of the maintainers that was involved with the issue.
-- If a reviewer commented on your code or asked for changes, please remember to mark the discussion as resolved after you address it and re-request a review.
-- When addressing comments, try to fix each suggestion in a separate commit.
-- Tests are not required at this point as Tracee is evolving fast, but if you can include tests that will be appreciated.
+1. Every Pull Request should have an associated Issue unless it is a trivial fix.
+1. When adding a flag option or other UX related change, make sure the design is explicitly described in the associated issue, and a maintainer approved it.
+1. Commit subject should succinctly describe the change. 
+1. Optional commit body (separated form subject by newline) can further explain what and why, not how.
+1. Commit subject style: 
+  1. Max 50 chars.
+  1. Phrased like a sentence.
+  1. Capitalized but doesn't end with period. 
+  1. Written in imperative mood, try to begin the sentence with either "Fix" or "Add".
+  1. Try to mention the area that this commit affects (see area labels above).
+1. Body style: 
+  1. Wrap at 72 chars.
+1. Code related information should be in commit message, review related information should be in PR description.
+
+## Code
+
+1. Follow Golang's code review standards: https://github.com/golang/go/wiki/CodeReviewComments.
+1. Follow `gofmt` + `govet` + `goimports` formatting.
+1. Tests should be considered an integral part of the PR. For parts that are harder to test, and are not currently tested, it's not required.
```

Comment:
Should we say all new code should have tests to go along with it unless it's impossible to test?